src/Symfony/Bridge/Doctrine/Repository/RepositoryTrait.php

```diff
@@ -0,0 +1,194 @@
+     */
+    public function createResultSetMappingBuilder($alias)
+    {
+        return $this->getRepository()->createResultSetMappingBuilder($alias);
+    }
+
+    /**
+     * @see EntityRepository::createNamedQuery()
+     *
+     * @param string $queryName
+     *
+     * @return Query
+     */
+    public function createNamedQuery($queryName)
+    {
+        return $this->getRepository()->createNamedQuery($queryName);
+    }
+
+    /**
+     * @see EntityRepository::createNativeNamedQuery()
+     *
+     * @param string $queryName
+     *
+     * @return NativeQuery
+     */
+    public function createNativeNamedQuery($queryName)
+    {
+        return $this->getRepository()->createNativeNamedQuery($queryName);
+    }
+
+    /**
+     * @see EntityRepository::clear()
+     */
+    public function clear()
+    {
+        $this->getRepository()->clear();
+    }
+
+    /**
+     * @see EntityRepository::find()
+     *
+     * @param mixed    $id          the identifier
+     * @param int|null $lockMode    one of the \Doctrine\DBAL\LockMode::* constants
+     *                              or NULL if no specific lock mode should be used
+     *                              during the search
+     * @param int|null $lockVersion the lock version
+     *
+     * @return object|null the entity instance or NULL if the entity can not be found
+     */
+    public function find($id, $lockMode = null, $lockVersion = null)
+    {
+        return $this->getRepository()->find($id, $lockMode, $lockVersion);
+    }
+
+    /**
+     * @see EntityRepository::findAll()
+     *
+     * @return array the entities
+     */
+    public function findAll()
+    {
+        return $this->getRepository()->findAll();
+    }
+
+    /**
+     * @see EntityRepository::findBy()
+     *
+     * @param array      $criteria
+     * @param array|null $orderBy
+     * @param int|null   $limit
+     * @param int|null   $offset
+     *
+     * @return array the objects
+     */
+    public function findBy(array $criteria, array $orderBy = null, $limit = null, $offset = null)
+    {
+        return $this->getRepository()->findBy($criteria, $orderBy, $limit, $offset);
+    }
+
+    /**
+     * @see EntityRepository::findOneBy()
+     *
+     * @param array      $criteria
+     * @param array|null $orderBy
+     *
+     * @return object|null the entity instance or NULL if the entity can not be found
+     */
+    public function findOneBy(array $criteria, array $orderBy = null)
+    {
+        return $this->getRepository()->findOneBy($criteria, $orderBy);
+    }
+
+    /**
+     * @see EntityRepository::matching()
+     *
+     * @param \Doctrine\Common\Collections\Criteria $criteria
+     *
+     * @return \Doctrine\Common\Collections\Collection
+     */
+    public function matching(Criteria $criteria)
+    {
+        return $this->getRepository()->matching($criteria);
+    }
+
+    /**
+     * @return EntityManagerInterface
+     */
+    protected function getEntityManager()
```

Comment:
getEntityManager and getRepository should be marked as final so we can return types in sf 4